### DISCUSSION TITLE
The Future of the Project (Joint Development)

### DISCUSSION THREAD

**Discussion by bigcat88** [cubiq/Mellon#2] [General]
Good time of day !  

I have a few questions about your project and its future direction.  

I’m deeply interested in diffusion models and everything related to LLMs. The idea of creating an analogue to ComfyUI/A1111 from scratch, while addressing past challenges, is incredibly exciting.  

We have a project called **Visionatrix**, where we currently use ComfyUI but aim to make it more user-friendly and support distributed flow execution (something akin to SwarmUI, but with a different approach).  

We’ve developed and tested a rough draft of the database schema, API endpoints, and Pydantic models under different environments and loads. While these are currently tailored for ComfyUI, they could be adapted for Mellon with minimal effort.  

Having seen your project, I’d like to ask:  

1. Would you be interested in creating a shared organization for collaboration? We could either establish a new org or rename the Visionatrix org—it’s flexible.  
2. Would you consider working together to combine efforts?  

By collaborating, we could support:  
- Two different UIs
- Two types of endpoints (HTTP + WebSocket/QUIC)
- A shared database for settings
- Common tests and infrastructure

You could focus on the core engine, and I’d be happy to assist where needed. While I have experience with Windmill (popular in government organizations), it’s not suitable for diffusion/transformers.  

### My Vision for the Project  

I’d like to leverage modern, stable tools, including:  
- **Starlette/FastAPI** and **Pydantic**  
- **OpenAPI** for clear and extensible documentation  
- Auto-generated UI support (in Visionatrix, I’ve developed a ChatGPT prompt to generate Gradio apps from OpenAPI specs, though this feature hasn’t been released yet, but it is working)  

I’m also working on agency/tool-calling features as part of my main job, which could integrate well with this project.  

#### Core Features:  
1. **Three Types of UI**:  
   - A standard flow engine like the one you’re building.  
   - A simplified UI (similar to Visionatrix), focused solely on usage, with editable node parameters but no add/delete functionality.
   - A conversational UI ("ChattyUI"), where users describe issues or goals, and an LLM with vision capabilities interprets tasks and uses agents/functions to achieve the desired results.  

2. Comprehensive task history support.  
3. Flexible user backends: simple options for home use, and LDAP support for production environments.  
4. Seamless scalability with remote GPUs/NPUs for task processing. Distributed execution is nearly ready but requires adaptation for Mellon.  
5. Automatic model support, including diffusers/transformers, with workflows that can automatically download required models.  

#### Optional Features:  
- **Plugin Support**: Hosted on PyPI as standard Python packages, with the backend providing a unified API for plugins.  
- **Simple Installation for Home Use**: Avoiding Electron, but offering a lightweight preconfiguration tool (similar to Skyrim/Fallout preconfig "ini" game tools but we can store all configs in the DB), written in GTK and packed as an exe/dmg for easy deployment of core components.  

I'm excited about the potential for collaboration and would love to hear your thoughts and maybe I missed something, cause did not gather all ideas that come into mind, just first ones which is troubling me currently.

**Comment by cubiq** [comment]
thanks for reaching out, at the moment the development is still too early and I don't know exactly where we I'm going with this. I have a lot of ideas, like an "appify" option and distributed inference across different hosts.

at the moment I'm mostly trying to understand what is the best way to interact with diffusers.

So, yes, of course I'm interested in a collaboration and let's keep in touch, but I need at least a v0.1 first :)

**Reply by bigcat88** [reply]
Great, I hope we won't have to wait too long for version 0.1 =)

We can create a chat somewhere in Discord or Telegram and discuss it, so as not to spam the repo.
Or, of course, we can create a separate discussions for each topic/idea separately, in case other collaborators appear or just smart people will give advice.

*For now, I won't do anything new that is heavily dependent on ComfyUI, so as not to rewrite it later.*

**Reply by cubiq** [reply]
contact me on discord if you want.

As per Comfy, at the moment... it's the sane option to follow, and it will be even more so in the future.
I just have different ideas and I'm in that kind of position of "because I can".
